fdbserver/workloads/KillRegion.actor.cpp

```diff
@@ -94,10 +101,13 @@ struct KillRegionWorkload : TestWorkload {
 		TraceEvent("ForceRecovery_GotConfig").detail("Conf", conf.toString());
 
 		if(conf.usableRegions>1) {
-			//only needed if force recovery was unnecessary and we killed the secondary
-			wait( success( changeConfig( cx, g_simulator.disablePrimary + " repopulate_anti_quorum=1", true ) ) );
-			while( self->dbInfo->get().recoveryState < RecoveryState::STORAGE_RECOVERED ) {
-				wait( self->dbInfo->onChange() );
+			loop {
+				//only needed if force recovery was unnecessary and we killed the secondary
+				wait( success( changeConfig( cx, g_simulator.disablePrimary + " repopulate_anti_quorum=1", true ) ) );
```

Comment:
Can you explain why we need to retry this `changeConfig` and why this can help have `waitForStorageRecovered ` returned?

It will be useful to briefly explain the reason as a comment in the code as well.

Reply:
There are corner cases where the data distributor will miss that the primary is dead and this will cause some undesired behavior. This change config in a loop is a workaround that will enable the data distributor algorithm to be restarted to pick up the knowledge that the primary is dead. This is an extreme corner case that is unlikely to happen in the real world. More details is available in the commit message.